javatests/arcs/core/storage/DirectStoreMuxerTest.kt

```diff
@@ -99,4 +159,221 @@ class DirectStoreMuxerTest {
+        val callbackId1 = directStoreMuxer.on(ProxyCallback {
+            assertThat(it is ProxyMessage.ModelUpdate).isTrue()
+            job.complete()
+        })
+
+        // other proxy muxer
+        var callbackInvoked = false
+        val callbackId2 = directStoreMuxer.on(ProxyCallback {
+            callbackInvoked = true
+        })
+
+        // set up store for muxId "a" and register for each proxy muxer
+        val muxIdA = "a"
+        val (idSet, store) = directStoreMuxer.store(muxIdA)
+        directStoreMuxer.registerToStore(callbackId1, muxIdA, idSet, store)
+        directStoreMuxer.registerToStore(callbackId2, muxIdA, idSet, store)
+
+        directStoreMuxer.onProxyMessage(ProxyMessage.SyncRequest(id = callbackId1, muxId = muxIdA))
+        job.join()
+        assertThat(callbackInvoked).isFalse()
+    }
+
+    @Test
+    fun unregisterSuccessfully() = runBlockingTest {
+        val driverProvider = MockDriverProvider()
+        DriverFactory.register(driverProvider)
+
+        val directStoreMuxer = createDirectStoreMuxer()
+
+        val callbackId1 = directStoreMuxer.on(ProxyCallback {})
+
+        val muxIdA = "a"
+        val (idSetA, storeA) = directStoreMuxer.store(muxIdA)
+        directStoreMuxer.registerToStore(callbackId1, muxIdA, idSetA, storeA)
+
+        val muxIdB = "b"
+        val (idSetB, storeB) = directStoreMuxer.store(muxIdB)
+        directStoreMuxer.registerToStore(callbackId1, muxIdB, idSetB, storeB)
+
+        assertThat(idSetA.size).isEqualTo(1)
+        assertThat(idSetB.size).isEqualTo(1)
+
+        directStoreMuxer.off(callbackId1)
+
+        assertThat(idSetA).isEmpty()
+        assertThat(idSetB).isEmpty()
+    }
+
+    // region Helpers
+    private fun createDirectStoreMuxer():
+        DirectStoreMuxer<CrdtEntity.Data, CrdtEntity.Operation, CrdtEntity> {
+        return DirectStoreMuxer<CrdtEntity.Data, CrdtEntity.Operation, CrdtEntity>(
+            storageKey,
+            backingType = EntityType(schema)
+        )
+    }
+
+    private fun DirectStoreMuxer<CrdtEntity.Data, CrdtEntity.Operation, CrdtEntity>.getEntityDriver(
+        id: ReferenceId
+    ): MockDriver<CrdtEntity.Data> =
+        requireNotNull(stores[id]).store.driver as MockDriver<CrdtEntity.Data>
+
+    private fun createPersonEntityCrdt(name: String, age: Int): CrdtEntity = CrdtEntity(
+        CrdtEntity.Data(
+            versionMap = VersionMap("me" to 1),
+            singletons = mapOf(
+                "name" to CrdtSingleton(
+                    initialVersion = VersionMap("me" to 1),
+                    initialData = CrdtEntity.Reference.buildReference(name.toReferencable())
+                ),
+                "age" to CrdtSingleton(
+                    initialVersion = VersionMap("me" to 1),
+                    initialData = CrdtEntity.Reference.buildReference(age.toReferencable())
+                )
+
+            )
+        )
+    )
+
+    // endregion
+
+    // region Mocks
```

Comment:
since this is duplicate from other tests, can you add a todo to refractor it into its own file?

Reply:
Done.